src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexAssert.netcoreapp.cs

```diff
@@ -15,7 +15,7 @@ public static class RegexAssert
     {
         public static void Equal(string expected, Capture actual)
         {
-            Assert.Equal(expected, actual.Value);
+            Assert.True(expected == actual.Value, $"Expected {Regex.Escape(expected)} actual {Regex.Escape(actual.Value)}");
```

Comment:
this makes the message more useful in cases when eg the value contains a newline.